hatchery/nextflow.go

```diff
@@ -1075,11 +1083,53 @@ func setupSubnet(subnetName string, cidr string, vpcid string, ec2Svc *ec2.EC2)
 		return exsubnet.Subnets[0].SubnetId, nil
 	}
 
+	// Fetch all availability zones
+	// this is being limited to a region by the ec2svc that gets passed in.
+	describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe availability zones: %v", err)
+	}
+
+	// Make sure the selected AZ has the instance type from nextflow configuration available.
+	var selectedZone string
+	for _, zone := range describeZonesOutput.AvailabilityZones {
+		if *zone.State == "available" {
```

Comment:
Instead of adding an `if` condition here checking for availability of an Availability Zone, can we add a filter in the above line? 
```
describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
```

Note: I haven't looked too deep into the docs, but if the input is used as a filter, maybe that is better than us filtering it ourselves.